From 9131bcf0790d926727a2866638c407451602c7f1 Mon Sep 17 00:00:00 2001
From: spoluri <sarat.chandra.poluri@intel.com>
Date: Thu, 9 May 2019 12:07:46 -0700
Subject: [PATCH 1094/1225] media: intel-ipu4: [ICI] Align with V4L2 driver
 behavior in DMA code when issue is detected.

Update ICI DMA logic to align with V4L2 driver for error detection

Change-Id: Iecd0f4179e00f87dc4f28c70fe5e2a763b1136eb
Tracked-On: PKT-2287
Tracked-On: HSDES-1409370304
Signed-off-by: spoluri <sarat.chandra.poluri@intel.com>
---
 drivers/media/pci/intel/ici/ici-dma.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/media/pci/intel/ici/ici-dma.c b/drivers/media/pci/intel/ici/ici-dma.c
index 430ece8..6b39bf2 100644
--- a/drivers/media/pci/intel/ici/ici-dma.c
+++ b/drivers/media/pci/intel/ici/ici-dma.c
@@ -263,7 +263,8 @@ static void intel_ipu4_dma_free(struct device *dev, size_t size, void *vaddr,
 	if (WARN_ON(!area->pages))
 		return;
 
-	BUG_ON(!iova);
+	if (WARN_ON(!iova))
+		return;
 
 	size = PAGE_ALIGN(size);
 
@@ -328,7 +329,8 @@ static void intel_ipu4_dma_unmap_sg(struct device *dev,
 	if (!nents)
 		return;
 
-	BUG_ON(!iova);
+	if (WARN_ON(!iova))
+		return;
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 8, 0)
 	if (!dma_get_attr(DMA_ATTR_SKIP_CPU_SYNC, attrs))
-- 
2.7.4

